### ISSUE TITLE
Ambience sound resets between screens

### ISSUE THREAD

**Issue by snouz** [Albeoris/Memoria#140]
Ambience sounds (i.e. Alexandrian, owl...) resets each time you change screen instead of blending.

**Comment by poepkat** [comment]
Hi,

Here are comparison videos. Sorry I can't contribute more, I'm not a programmer.

Bugged loop (list to the owl sounds, the start immediately after every screen change): https://streamable.com/pba6bx
Proper loop (I don't change screens so you can hear what the loop is supposed to sound like): https://streamable.com/k3bh1r

**Comment by poepkat** [comment]
Hi,

Sorry is this is the wrong place to post noob questions. If you don't want to answer I understand. I've checked out the code you linked, and it got me intrigued to try and understand it a little bit.

First of all, how would one know StopAllSoundEffects are related to ambient soundtracks specifically? 

Second
```
			if (this.CurrentScene != this.NextScene)
			{
				if (String.Equals(this.CurrentScene, "QuadMist"))
				{
					SceneDirector.FF9Wipe_FadeInEx(30);
				}
				if (String.Equals(this.NextScene, "MainMenu"))
				{
					SoundLib.StopAllSounds(true);
				}
etc. etc.
```

The he game calls StopAllSoundEffects if the current scene is not the same as the next scene ONLY IF any of the conditions of the string are met (so any of the code that comes AFTER if (this.CurrentScene != this.NextScene) or does it ALWAYS execute StopAllSoundEffects on EVERY screen change? 

And as as follow op, am I reading this correctly: if the current scene is QuadMist (what is that, the Game Over screen or something?) and it's not the same as the previous scene, the game starts FF9_Wipe_FadeInEx, or of the next scene is MainMenu then SoundLib.StopAllSounds is executed, which probably disables both SoundEffects and the Music?

Cheers,

**Comment by Tirlititi** [comment]
To answer your questions, Poepkat, the "ChangeScene" triggers on every screen change (except opening the menus and fake screen changes, like Iifa Tree roots for which several screens are actually the same field). QuadMist is the old/japanese name of Tetra Master. The "this.CurrentScene != this.NextScene" condition doesn't check if the screen changes but if the next module is the same as the previous module (is it a field, a world map, a battle, a Tetra Master game, a game over...).
"StopAllSoundEffects" doesn't apply to ambiant soundtracks only ("SNDEFFECTRES") but only them (and possibly songs) would be problematic because they would be never-ending if not handled properly.

I have not tested the fix I just did thoroughly in-game but that seems to work fine until at least disc 2.

**Comment by poepkat** [comment]
Would it be possible for you to provide a Assembly-CSharp.dll with the Ambience fix (#140?) that is compatible with Moguri/Alternate Fantasy; I'm assuming I can just replace the Assembly-CSharp.dll in the Steam game folder? I'm in a place with very bad internet currently so I cannot download any big programmes to (learn to) compile myself, but tomorrow I have a long flight home and I would like to play FF9 in the plane. Or can you tell me what I should do to apply the fix to Moguri/Alternate Fantasy?
